console/console-init/ui/src/modules/iot-device/utils/TableFormatter.tsx

```diff
@@ -38,8 +38,11 @@ export const getTableCells = (row: IDevice, selectedColumns: string[]) => {
   } = row;
 
   let deviceType: DeviceConnectionType;
-
-  if (via?.length || viaGroups?.length) {
+  const viaGateway =
+    (via && via.length > 0) || (viaGroups && viaGroups.length > 0);
+  if (!via?.length === !credentials?.length) {
```

Comment:
This condition looks confusing and you need to check via and viaGroups both.
it can be.
if (!viaGateway &&  !credentials?.length) {
  deviceType = DeviceConnectionType.NA;
}